package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -184,101 +137,234 @@ class as the topology and coordinate reader share many common
+        header: dict of str
+            A dict in which a key and value pair is a 'header' name and
+            its value in str foramt.
+        sections: dict of list of str
+            A dict in which a key and value pair is a 'section' name and
+            a list of strings (relevenat lines of the data file).
         """
         f = list(self.iterdata())
 
+        # finding the starting line of each section:
         starts = [i for i, line in enumerate(f)
-                  if line.split()[0] in SECTIONS]
+                  if line.split()[0] in self.SECTIONS]
+        # adding None to properly handle lines in the last SECTION below
         starts += [None]
 
         header = {}
+
+        # lines before the first SECTION found in the file are about HEADERS:
         for line in f[:starts[0]]:
-            for token in HEADERS:
+            for token in self.HEADERS:
                 if line.endswith(token):
                     header[token] = line.split(token)[0]
                     continue
 
-        sects = {f[l]:f[l+1:starts[i+1]]
+        sects = {f[l]: f[l+1:starts[i+1]]  # f[l] is indeed a SECTION name
                  for i, l in enumerate(starts[:-1])}
 
         return header, sects
 
     @staticmethod
     def _interpret_atom_style(atom_style):
-        """Transform a string description of atom style into a dict
-
-        Required fields: id, type, x, y, z
-        Optional fields: resid, charge
-
-        eg: "id resid type charge x y z"
-        {'id': 0,
-         'resid': 1,
-         'type': 2,
-         'charge': 3,
-         'x': 4,
-         'y': 5,
-         'z': 6,
-        }
-        """
-        style_dict = {}
+        """Map field names in a given `atom_style` to their corresponding
+        locations.
 
-        atom_style = atom_style.split()
+        Required attributes: id, type, x, y, z
+        Optional attributes: resid, charge
+        Note: other attributes are ignored.
 
-        for attr in ['id', 'type', 'resid', 'charge', 'x', 'y', 'z']:
-            try:
-                location = atom_style.index(attr)
-            except ValueError:
-                pass
-            else:
-                style_dict[attr] = location
-
-        reqd_attrs = ['id', 'type', 'x', 'y', 'z']
-        missing_attrs = [attr for attr in reqd_attrs if attr not in style_dict]
-        if missing_attrs:
-            raise ValueError("atom_style string missing required field(s): {}"
-                             "".format(', '.join(missing_attrs)))
-                
+        Exammples
+        ---------
+        "full" atom_style has the following attributes:
+            "id resid type charge x y z"
+        `_interpret_atom_style` returns this dict for "full" atom_style:
+            {'id': 0,
+             'resid': 1,
+             'type': 2,
+             'charge': 3,
+             'x': 4,
+             'y': 5,
+             'z': 6,
+            }
+
+        Arguments
```

Comment:
`Arguments`  is not a valid numpy key word IIRC it must be 
```
Parameters
----------
```
This applies to other ones also.